### DISCUSSION TITLE
What's the best way to handle configuriatons for different enviornments

### DISCUSSION THREAD

**Discussion by knightsrule** [quasarframework/quasar#17630] [CLI - PWA mode]
I would like to use development enviornment settings when running "quasar dev" or when deploying to a dev stack in AWS. I'd like ot use prodeuction configuration settings when building for production. If i add a "mode" flag to "quasar build mode=dev/prod", "quasar dev" doesn't pick this flag up. 

What's the best way to handle this.

I tried this page as a guidance, https://quasar.dev/quasar-cli-vite/quasar-config-file, but I have two issues:
1. My project doesn't have a quasar.config and manually adding it doesn't do much
2. this doesn't show to build for dev configuration.

Thanks!

**Comment by yusufkandemir** [comment]
The term `quasar.config file` represents `quasar.config.js`/`quasar.config.ts`.

What you want is to use env variables for configuration, see:
https://quasar.dev/quasar-cli-vite/handling-process-env

**Comment by knightsrule** [comment]
Thanks. How do I build using Dev configuration ? Checking Node_env’s value will work but I don’t know how to impact its value. Quasar build uses production configuration Sent from my iPhoneOn Nov 9, 2024, at 7:15 AM, Yusuf Kandemir ***@***.***> wrote:﻿
The term quasar.config file represents quasar.config.js/quasar.config.ts.
What you want is to use env variables for configuration, see:
https://quasar.dev/quasar-cli-vite/handling-process-env

**Reply by yusufkandemir** [reply]
Have you read the page I've sent? Read it, this section will clarify things for you: https://quasar.dev/quasar-cli-vite/handling-process-env#adding-to-process-env
you can also use dotenv as can be seen there
